### DISCUSSION TITLE
Problem with crop a mask

### DISCUSSION THREAD

**Discussion by 22bco** [ultralytics/ultralytics#1243] [Q&A]
I need to crop the outline of the image with the coordinates of the segment, the mask... is there a function inside YOLOv8 and utils to do it instead of doing manual? Similar to the function that generates the crop and saves it --save_crop

I was seeing the docs and ultralytics/yolo/utils/ops.py crop_mask functions but It doesn't work

**Comment by Plogeur** [comment]
same issue if you got the answer let me know pls.

**Reply by glenn-jocher** [reply]
@Plogeur hello! In YOLOv8, while we do have utilities for various operations, there isn't a direct function named `crop_mask` that you can use to crop the outline of an image with the coordinates of a segment. The `--save_crop` functionality is specific to saving cropped detections, not segmentations.

For segmentation tasks, you would typically obtain a binary mask that corresponds to the segmented area. To crop this area, you would need to use the mask to extract the region of interest from the image. This is generally done by applying the mask to the image and then using bounding box coordinates to crop it.

If you're looking to work with segmentation masks and need to crop them, you might need to write a custom function that takes the mask and the image as input and then performs the cropping operation. The process usually involves finding contours in the mask, determining the bounding box of the contour, and then using these coordinates to crop the image.

For more details on how to handle different computer vision tasks with YOLOv8, including segmentation, please refer to our documentation on segmentation tasks. If you're looking for more specific guidance on handling masks and cropping, you might find useful information under the detection tasks section, as the principles for handling bounding boxes can often be adapted for segmentation masks.

Remember, the YOLO community and the Ultralytics team are always working to improve and add features, so keep an eye on our documentation and updates for any new utilities that might help with your task. 😊
